File: Updog.Domain/Entities/Post/Post.cs
using System;

namespace Updog.Domain {
    /// <summary>
    /// A post of the website.
    /// </summary>
    public sealed class Post : IEntity {
        #region Constants
        /// <summary>
        /// The max # of characters allowed in a post title.
        /// </summary>
        public const int TitleMaxLength = 300;

        /// <summary>
        /// The max # of characters allowed in a post body.
        /// </summary>
        public const int BodyMaxLength = 10_000;
        #endregion

        #region Properties
        /// <summary>
        /// Unique numeric Id.
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// OP of the post.
        /// </summary>
        public User User { get; set; }

        /// <summary>
        /// The content type of the post.
        /// </summary>
        public PostType Type { get; set; }

        /// <summary>
        /// The title of the post.
        /// </summary>
        /// <value></value>
        public string Title { get; set; }

        /// <summary>
        /// The body of the post. Either a URL or text body.
        /// </summary>
        public string Body { get; set; }

        /// <summary>
        /// When the post was created.
        /// </summary>
        /// <value></value>
        public DateTime CreationDate { get; set; }

        /// <summary>
        /// If the post was editted by a user.
        /// </summary>
        public bool WasUpdated { get; set; }

        /// <summary>
        /// If the post was deleted by a user.
        /// </summary>
        /// <value></value>
        public bool WasDeleted { get; set; }

        /// <summary>
        /// The comments of the post.
        /// </summary>
        public Comment[] Comments { get; set; }
        #endregion
    }
}

File: Updog.Client/src/post/domain/post.ts
import { PostType } from './post-type';
import { User } from '@/user/domain/user';
import { PaginationParams } from '@/core/pagination/pagination-params';
import { UserEntity } from '@/user/common/user-entity';
import { Vote } from '@/vote/domain/vote';
import { Space } from '@/space/domain/space';
import { VoteDirection } from '@/vote/domain/vote-direction';
import { VoteResourceType } from '@/vote/domain/vote-resource-type';
import { VotableEntity } from '@/vote/common/votable-entity';

/**
 * Post made by a user. Probably a repost...
 */
export class Post extends VotableEntity {
    /**
     * Default number of posts in a page.
     */
    public static DEFAULT_PAGE_SIZE: number = 20;

    /**
     * Max char count for the title.
     */
    public static TITLE_MAX_LENGTH: number = 300;

    /**
     * Max char count for the body.
     */
    public static BODY_MAX_LENGTH: number = 10_000;

    /**
     * Flag to indicate what kind of votable entity it is.
     */
    public voteResourceType: VoteResourceType = VoteResourceType.Post;

    /**
     * Create a new post.
     * @param id The unique ID of the post.
     * @param type THe type of post it is.
     * @param title The title of the post.
     * @param body The body of the post.
     * @param user The user that posted the post.
     * @param space The space it was posted to.
     * @param creationDate Date and time of posting.
     * @param commentCount The number of comments on it.
     * @param wasUpdated If the post was editted.
     * @param wasDeleted If the post was deleted.
     * @param upvotes The number of upvotes it go.
     * @param downvotes The number of downvotes it has.
     * @param vote The vote cast by the current user.
     */
    constructor(
        public id: number,
        public type: PostType,
        public title: string,
        public body: string,
        public user: User,
        public space: Space,
        public creationDate: Date,
        public commentCount: number,
        public wasUpdated: boolean,
        public wasDeleted: boolean,
        public upvotes: number,
        public downvotes: number,
        public vote: Vote | null = null
    ) {
        super();
    }
}
